src/Fmt_ast.ml

```diff
@@ -1449,43 +1506,6 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
   @@
   match pexp_desc with
   | Pexp_apply (_, []) -> impossible "not produced by parser"
-  | Pexp_sequence
-      ( { pexp_desc=
-            Pexp_extension
-              ( name
-              , PStr
-                  [ ( { pstr_desc=
-                          Pstr_eval
-                            ( ( { pexp_desc= Pexp_fun _
-                                ; pexp_loc= _
-                                ; pexp_attributes= _
-                                ; _ } as call )
-                            , [] )
-                      ; pstr_loc= _ } as pld ) ] )
-        ; pexp_attributes= _
-        ; pexp_loc= _
-        ; _ }
-      , e2 ) ->
-      let xargs, xbody = Sugar.fun_ c.cmts (sub_exp ~ctx:(Str pld) call) in
-      let is_simple x = is_simple c.conf width x in
-      let break xexp1 xexp2 = not (is_simple xexp1 && is_simple xexp2) in
-      let grps =
-        List.group (Sugar.sequence c.cmts (sub_exp ~ctx e2)) ~break
-      in
-      let fmt_grp grp = list grp " ;@ " (fmt_expression c) in
-      hvbox 0
-        (wrap_if parens "(" ")"
-           ( hvbox c.conf.extension_indent
-               (wrap "[" "]"
-                  ( str "%"
-                  $ hovbox 2
-                      ( fmt_str_loc c name $ str " fun "
-                      $ fmt_attributes c ~suf:(str " ") call.pexp_attributes
-                          ~key:"@"
-                      $ fmt_fun_args c xargs $ fmt "@ ->" )
-                  $ fmt "@ " $ fmt_expression c xbody ))
-           $ fmt "@ ;@ "
-           $ list grps " ;@;<1000 0>" fmt_grp ))
```

Comment:
I removed this special case, I don't see what made it necessary, before we had:
```ocaml
let _ =
  [%foooo fun foooooooooooooooooooo fooooooooooooooo fooo -> fooooooooo]
  ;
  foooooooooooooooooooooooo
```
now we have:
```ocaml
let _ =                
  [%foooo fun foooooooooooooooooooo fooooooooooooooo fooo -> fooooooooo] ;
  foooooooooooooooooooooooo
```
and it benefits from the changes of this PR.

Reply:
Looks good.